lib/src/country_flags.dart

```diff
@@ -92,4 +94,47 @@ class CountryFlag extends StatelessWidget {
       ),
     );
   }
+
+  /// Build the flag widget according to the specified shape.
+  Widget _buildFlagWidget() {
+    switch (shape) {
```

Comment:
This is a good opportunity to get advantage of the new pattern matching feature introduced in Dart 3.0:
```
switch (shape) {
   Shape.circle => const _buildCircularFlag(),
   Shape.rectangle => const _buildRectangularFlag(),
},
```